table/server/master/src/main/java/alluxio/master/table/Table.java

```diff
@@ -86,6 +86,8 @@ private Table(Database database, UdbTable udbTable, @Nullable Table previousTabl
           previousTable.mPartitionScheme.getPartitions().stream()
               .collect(Collectors.toMap(Partition::getSpec, Function.identity()));
       for (UdbPartition udbPartition : udbTable.getPartitions()) {
+        LOG.debug("Get partition: {}, from table {} Database {}.",
+            udbPartition.toString(), database.getName(), mName);
         Partition newPartition = existingPartitions.get(udbPartition.getSpec());
         if (newPartition == null) {
```

Comment:
I would suggest being able to distinguish the 3 cases here. Something like the following. This would allow us to reall know which code path/scenario each partition falls under, and would help with debugging.

Also, using `if (LOG.isDebugEnabled())` since `toString` could be expensive.

```
if (newPartition == null) {
  // partition does not exist yet
  newPartition = new Partition(udbPartition);
  if (LOG.isDebugEnabled()) {
    LOG.debug("Existing table {}.{} adding UDB partition: {}", ...);
  }
} else if (!newPartition.getBaseLayout().equals(udbPartition.getLayout())) {
  // existing partition is updated
  newPartition = newPartition.createNext(udbPartition);
  if (LOG.isDebugEnabled()) {
    LOG.debug("Existing table {}.{} updating UDB partition: {}, ...");
  }
} else {
  if (LOG.isDebugEnabled()) {
    LOG.debug("Existing table {}.{} keeping partition spec: {}", ...);
  }
}
```

Reply:
yeah, it's much more efficient, changed.